go/arrow/csv/reader.go

```diff
@@ -507,6 +511,23 @@ func (r *Reader) parseFloat64(field array.Builder, str string) {
 	field.(*array.Float64Builder).Append(v)
 }
 
+// parses timestamps using millisecond precision
+func (r *Reader) parseTimestamp(field array.Builder, str string) {
+	if r.isNull(str) {
+		field.AppendNull()
+		return
+	}
+
+	v, err := arrow.TimestampFromString(str, arrow.Millisecond)
```

Comment:
We should probably get the unit either from a Reader option or a given Schema instead of assuming millisecond here.

Reply:
@zeroshade great suggestion, I have added the option.